types/src/blob.rs

```diff
@@ -52,6 +52,20 @@ pub struct Blob {
     pub share_version: u8,
     /// A [`Commitment`] computed from the [`Blob`]s data.
     pub commitment: Commitment,
+    #[serde(deserialize_with = "deserialize_optional_positive_integer")]
+    pub index: Option<u64>,
+}
+
+fn deserialize_optional_positive_integer<'de, D>(deserializer: D) -> Result<Option<u64>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let value: i64 = Deserialize::deserialize(deserializer)?;
```

Comment:
nit

```suggestion
    let value = i64::deserialize(deserializer)?;
```

Reply:
@S1nus this one seems to be overwritten by later commits. I can't spot this change in diff